### PR TITLE
Task/dat 1226 use signed int

### PR DESCRIPTION
**Pull request by hb0**
I also upgraded the dependencies. If I should split this PR into two PRs, let me know.

**Breaking Changes**
- `uint64` --> `sint64` of LocationRecords, Accelerations, Rotations, Directions `timestamp`s
- `uint` and `sint` are not compatible
- instead of using new field numbers for the message (backend-compatibility) we decided to increase the binary version to 3 to keep it simple as there was no reason why we need to support backward compatibility to version 2.

### CODE REVIEW

File: src/main/proto/de/cyface/protos/model/measurement.proto [hunk 1]
```diff
@@ -34,19 +34,17 @@ option java_multiple_files = true;
...
  */
 message Measurement {
   /*
    * The API version of this `Measurement` message format. Messages with the same API-version are compatible.
    *
-   * Using `32bit` as we expect the version to be and within the range of an `integer`.
+   * Using `uint32` as this is the smallest number type in Protobuf and we expect the versions to be between 0 and 2^32.
    *
-   * The value is encoded in the `uint` ProtoBuf format as we expect the version to be positive.
-   * Negative numbers are supported but encoded inefficiently in `uint` this data type, compared to `sint`.
```
Comment (hb0) [inline src/main/proto/de/cyface/protos/model/measurement.proto:47]: I removed or updated those wrong statements that `uint` support negative numbers.

File: src/main/proto/de/cyface/protos/model/measurement.proto [hunk 2]
```diff
@@ -239,10 +252,10 @@ message LocationRecords {
    *
    * Using `64bit` as the first timestamp is absolute and in milliseconds since 1970 and, thus, not an `integer`.
    *
-   * The value is encoded in the `uint` ProtoBuf format as we expect the timestamps to be in order (ascending).
-   * Negative numbers are supported but encoded inefficiently in `uint` this data type, compared to `sint`.
+   * The value is encoded in the `sint` ProtoBuf format to support cases where data points are collected in random order.
+   * Negative numbers are encoded more efficiently in this data type, compared to `int`.
    */
-  repeated uint64 timestamp = 1;
+  repeated sint64 timestamp = 1;
```
Comment (hb0) [inline src/main/proto/de/cyface/protos/model/measurement.proto:258]: That's the relevant fix (1 of 4).

File: src/main/proto/de/cyface/protos/model/measurement.proto [hunk 3]
```diff
@@ -366,10 +379,10 @@ message Accelerations {
...
```
Comment (hb0) [inline src/main/proto/de/cyface/protos/model/measurement.proto:385]: That's the relevant fix (2 of 4).

File: src/main/proto/de/cyface/protos/model/measurement.proto [hunk 4]
```diff
@@ -510,10 +523,10 @@ message Directions {
...
```
Comment (hb0) [inline src/main/proto/de/cyface/protos/model/measurement.proto:529]: That's the relevant fix (4 of 4).

File: src/main/proto/de/cyface/protos/model/measurement.proto [hunk 5]
```diff
@@ -584,8 +597,7 @@ message Event {
    *
    * Using `64bit` as the timestamp is absolute and in milliseconds since 1970 and, thus, not in the `integer` range.
    *
-   * The value is encoded in the `uint` ProtoBuf format as we don't expect negative timestamps.
-   * Negative numbers are supported but encoded inefficiently in `uint` this data type, compared to `sint`.
+   * The value is encoded in the `uint` ProtoBuf format as we don't expect (and support) negative absolute timestamps.
    */
   uint64 timestamp = 1;
```
Comment (hb0) [inline src/main/proto/de/cyface/protos/model/measurement.proto:602]: For non-`repeated` messages like `Events` we store only absolute `timestamp`s, i.e. we don't support negative numbers like in the `repeated` messages like `LocationRecords` and `Accelerations` where we store timestamp offsets and, thus, have to support negative numbers, too.

File: src/main/proto/de/cyface/protos/model/measurement.proto [hunk 6]
```diff
@@ -656,8 +668,7 @@ message File {
    *
    * Using `64bit` as the timestamp is absolute and in milliseconds since 1970 and, thus, not in the `integer` range.
    *
-   * The value is encoded in the `uint` ProtoBuf format as we don't expect negative timestamps.
-   * Negative numbers are supported but encoded inefficiently in `uint` this data type, compared to `sint`.
+   * The value is encoded in the `uint` ProtoBuf format as we don't expect (and support) negative absolute timestamps.
    */
   uint64 timestamp = 16;
```
Comment (hb0) [inline src/main/proto/de/cyface/protos/model/measurement.proto:673]: same as in `Events`  --> non `releated` --> we only store `absolute` timestamps.

**Review by hb0 — COMMENTED** [review]
pre-reviewed.

File: README.adoc [hunk 7]
```diff
@@ -62,20 +62,20 @@ However, the pre-compiled `JARs` are also published to the link:https://github.c
...
 link:https://developers.google.com/protocol-buffers/docs/javatutorial#builders[Source]
 
 
 [#_licensing]
 == Licensing
 
-Copyright (C) 2021 Cyface GmbH - All Rights Reserved
+Copyright (C) 2021-2022 Cyface GmbH - All Rights Reserved
 Unauthorized copying of this file, via any medium is strictly prohibited
 Proprietary and confidential
```
Comment (muthenberg) [inline README.adoc:81]: The Repo is public, so this is wrong.

**Review by muthenberg — CHANGES_REQUESTED** [review]
It would be cleaner to separate version changes and data type update into to PRs. If this is much of a hassle leave it, as it is now. But maybe remember for the future.

**Pull request merged**